File: static/CWE-502/python/vulnerable.py
# CWE-502: Deserialization of Untrusted Data
# Untrusted bytes are deserialized via pickle and YAML, enabling arbitrary code execution.
# Vulnerable sink: pickle.loads() and yaml.load() with an unsafe Loader

import pickle
import yaml
from flask import Flask, request

app = Flask(__name__)


@app.route("/load", methods=["POST"])
def load():
    data = request.get_data()
    obj = pickle.loads(data)
    config = yaml.load(request.args.get("cfg"), Loader=yaml.Loader)
    return str(obj) + str(config)

File: static/CWE-502/javascript/vulnerable.js
// CWE-502: Deserialization of Untrusted Data
// Untrusted request body is deserialized with node-serialize, allowing code execution via IIFE payloads.
// Vulnerable sink: serialize.unserialize(req.body.data)

const express = require('express');
const serialize = require('node-serialize');

const app = express();
app.use(express.json());

app.post('/load', (req, res) => {
  const data = req.body.data;
  const obj = serialize.unserialize(data);
  res.json({ loaded: obj });
});

app.listen(3000);
